beacon-chain/rpc/eth/validator/handlers.go

```diff
@@ -111,3 +110,52 @@ func (s *Server) SubmitContributionAndProofs(w http.ResponseWriter, r *http.Requ
 		}
 	}
 }
+
+// SubmitAggregateAndProofs verifies given aggregate and proofs and publishes them on appropriate gossipsub topic.
+func (s *Server) SubmitAggregateAndProofs(w http.ResponseWriter, r *http.Request) {
+	if r.Body == http.NoBody {
+		http2.HandleError(w, "No data submitted", http.StatusBadRequest)
+		return
+	}
+
+	var req SubmitAggregateAndProofsRequest
+	if err := json.NewDecoder(r.Body).Decode(&req.Data); err != nil {
+		http2.HandleError(w, "Could not decode request body: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	genesisTime := s.TimeFetcher.GenesisTime()
+
+	broadcastFailed := false
+	validate := validator.New()
+	for _, item := range req.Data {
```

Comment:
maybe you validate the req.Data here

Reply:
Good point. We should validate the whole `re` at the beginning.